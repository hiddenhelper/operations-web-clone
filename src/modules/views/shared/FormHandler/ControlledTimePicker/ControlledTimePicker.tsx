import React, { memo, useCallback } from 'react';
import { MuiPickersUtilsProvider, KeyboardTimePicker } from '@material-ui/pickers';
import MomentUtils from '@date-io/moment';

import { CalendarIcon } from '../../../../../constants';

export interface IControlledTimePickerProps {
  name: string;
  value: string;
  variant: 'standard' | 'outlined' | 'filled';
  iconPosition?: 'start' | 'end' | null;
  startDate?: string;
  endDate?: string;
  autoAccept?: boolean;
  placeholder?: string;
  styleClass?: string;
  inputProps?: any;
  fullWidth?: boolean;
  error: boolean;
  onChange: (data: any) => void;
  onOpen?: () => void;
  onAccept?: () => void;
  onClose?: () => void;
  onFocus?: () => void;
  onBlur?: () => void;
}

const ControlledTimePicker = ({
  name,
  value,
  variant,
  iconPosition,
  autoAccept = true,
  placeholder = '',
  styleClass = '',
  error,
  fullWidth,
  onChange,
  ...rest
}: IControlledTimePickerProps) => {
  const onDateChange = useCallback(
    date => {
      onChange({
        persist: /* istanbul ignore next */ () => {
          /* */
        },
        target: {
          name: name,
          value: date,
        },
      });
    },
    [onChange, name]
  );
  return (
    <MuiPickersUtilsProvider utils={MomentUtils}>
      <KeyboardTimePicker
        data-testid="keyboard-time-picker"
        format="hh:mm"
        value={value}
        onChange={onDateChange}
        inputVariant={variant}
        variant="inline"
        className={styleClass}
        keyboardIcon={<CalendarIcon />}
        InputAdornmentProps={{ position: iconPosition ? iconPosition : 'start' }}
        autoOk={autoAccept}
        placeholder={placeholder}
        error={error}
        fullWidth={fullWidth}
        {...rest}
      />
    </MuiPickersUtilsProvider>
  );
};

export default memo(ControlledTimePicker);
