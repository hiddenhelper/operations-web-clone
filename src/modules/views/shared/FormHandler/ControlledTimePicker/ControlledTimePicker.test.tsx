import React from 'react';
import { render, fireEvent, act } from '@testing-library/react';

import ControlledTimePicker, { IControlledTimePickerProps } from './ControlledTimePicker';

jest.mock('@material-ui/pickers', () => ({
  MuiPickersUtilsProvider: ({ children }) => children,
  KeyboardTimePicker: ({ onChange, value }) => {
    return (
      <div>
        <input data-testid="datepicker-item" value={value} onChange={onChange} />
      </div>
    );
  },
}));

describe('ControlledTimePicker', () => {
  let props: IControlledTimePickerProps;

  beforeEach(() => {
    props = {
      name: 'date-picker',
      value: '2020/10/10',
      variant: 'standard',
      error: false,
      onChange: jest.fn(),
    };
  });

  it('should render', () => {
    props.iconPosition = 'end';
    const wrapper = render(<ControlledTimePicker {...props} />);
    expect(wrapper.container).toMatchSnapshot();
  });

  it('should onChange', () => {
    const wrapper = render(<ControlledTimePicker {...props} />);

    act(() => {
      fireEvent.change(wrapper.getByTestId('datepicker-item'), { target: { value: '2020/10/12' } });
    });

    expect(props.onChange).toHaveBeenCalled();
  });
});
