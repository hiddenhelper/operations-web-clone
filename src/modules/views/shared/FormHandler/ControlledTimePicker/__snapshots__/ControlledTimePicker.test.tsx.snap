// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlledTimePicker should render 1`] = `
<div>
  <div>
    <input
      data-testid="datepicker-item"
      value="2020/10/10"
    />
  </div>
</div>
`;
